packages/design-toolkit/src/components/coordinate-field/index.tsx

```diff
@@ -0,0 +1,464 @@
+ * systems (DD, DDM, DMS, MGRS, UTM). All values are normalized to Decimal Degrees internally
+ * for consistency.
+ *
+ * @example
+ * // Basic coordinate field
+ * <CoordinateField label="Location" />
+ *
+ * @example
+ * // Coordinate field with validation
+ * <CoordinateField
+ *   label="Target Coordinates"
+ *   isRequired
+ *   isInvalid={hasError}
+ *   errorMessage="Please enter a valid coordinate"
+ * />
+ *
+ * @example
+ * // Coordinate field with specific format
+ * <CoordinateField
+ *   label="Position"
+ *   format="dms"
+ *   description="Enter coordinates in Degrees Minutes Seconds format"
+ * />
+ *
+ * @example
+ * // Compact coordinate field
+ * <CoordinateField
+ *   label="Coordinates"
+ *   size="small"
+ *   format="dd"
+ * />
+ *
+ * @example
+ * // Controlled coordinate field
+ * <CoordinateField
+ *   label="Selected Location"
+ *   value={coordinates}
+ *   onChange={setCoordinates}
+ * />
+ *
+ * @example
+ * // Coordinate field with error handling
+ * <CoordinateField
+ *   label="Target Coordinates"
+ *   onError={(message, context) => {
+ *     // message will be "Invalid coordinate value" for validation errors
+ *     // or "Invalid coordinate format" for paste errors
+ *     setErrorMessage(message);
+ *     console.error(message, context);
+ *   }}
+ *   isInvalid={!!errorMessage}
+ *   errorMessage={errorMessage}
+ * />
+ */
+export function CoordinateField({ ref, ...props }: CoordinateFieldProps) {
+  [props, ref] = useContextProps(props, ref, CoordinateFieldContext);
+
+  const {
+    classNames,
+    description: descriptionProp,
+    label: labelProp,
+    format = 'dd',
+    size = 'medium',
+    enableCopy = true,
+    isDisabled = false,
+    isInvalid: isInvalidProp = false,
+    isRequired = false,
+    ...rest
+  } = props;
+
+  const isSmall = size === 'small';
+
+  const DOMProps = filterDOMProps(rest, { global: true });
+
+  // When size is small and label is hidden, use aria-label instead of aria-labelledby
+  // to ensure screen readers have an accessible name
+  const ariaLabelForSmallSize =
+    isSmall && labelProp ? labelProp : rest['aria-label'];
+
+  const {
+    state,
+    focus,
+    paste,
+    copy,
+    fieldProps,
+    labelProps,
+    descriptionProps,
+    errorProps,
+    validation,
+    effectiveErrorMessage,
+    isInvalid,
+  } = useCoordinateField(
+    props,
+    ariaLabelForSmallSize,
+    rest['aria-describedby'],
+    rest['aria-details'],
+  );
+
+  const componentState = {
+    segmentValues: state.segmentValues,
+    format,
+    currentValue: state.currentValue,
+    validationErrors: state.validationErrors,
+    isDisabled,
+    isInvalid,
+    isRequired,
+    size,
+  };
+
+  // Memoize all coordinate format conversions to avoid redundant calculations
+  const allCoordinateFormats = useMemo(
+    () => getAllCoordinateFormats(state.currentValue),
+    [state.currentValue],
+  );
```

Comment:
option: because this is only used within the popover component which is only opened sometimes, you could move this to generate when the popover is opened to avoid extraneous calculations:

```suggestion
const [allCoordinateFormats, setAllCoordinateFormats] = useState(null);
const handlePopoverOpen = useCallback(() => {
  setAllCoordinateFormats(getAllCoordinateFormats(state.currentValue));
}, [state.currentValue]);
```